### DISCUSSION TITLE
Encountering error while trying to upgrade from 3.3.3 to 3.6.10

### DISCUSSION THREAD

**Discussion by Sagar-Singari** [belaban/JGroups#544] [Q&A] [answered]
Hi @belaban ,

I'm trying to upgrade from jgroups from3.3.3 final to 3.6.10 final but I'm encountering the following error: org.jgroups.protocols.TP handleMessageBatch SEVERE: JGRP000030: <hostname>: failed handling incoming message: java.lang.ArrayIndexOutOfBoundsException: -1

I'm aware that this issue has already been resolved and I need to use SYM_ENCRYPT instead of ENCRYPT, but the problem is I couldn't find where to make the changes. Can you point me in the right direction please.

Thanks.

**Comment by belaban** [comment, marked as answer]
What do you mean by "the issue has been resolved"? Which issue?

**Reply by Sagar-Singari** [reply]
The issue is the error: "**org.jgroups.protocols.TP handleMessageBatch SEVERE: JGRP000030: : failed handling incoming message: java.lang.ArrayIndexOutOfBoundsException: -1**" that I'm getting while trying to update from 3.3.3 to 3.6.10

And it was addressed in: https://issues.redhat.com/browse/JGRP-2088
I don't have an xml file to make the change but I see that I'm using org.jgroups.protocols.ENCRYPT, so I'm replacing it with SYM_ENCRYPT. Is that the only change I need to make to fix this error?

**Reply by Sagar-Singari** [reply]
This issue was resolved after I replaced ENCRYPT protocol with SYM_ENCRYPT

**Reply by belaban** [reply]
OK, good to know...
